Add public IP address allowlist for Teleport Cloud (#42593)
gravitational/teleport: docs/pages/choose-an-edition/teleport-cloud/ips.mdx

---
title: Public IP Address Allowlist
description: Restrict outbound network connections from your infrastructure to cloud-hosted Teleport Enterprise.
tocDepth: 3
---

> Teleport Agents connected to Teleport Enterprise Cloud must be allowed to connect to the following IP addresses in order to join the Teleport cluster.

Reviewer: ```suggestion
Teleport Agents connected to managed Teleport Enterprise accounts must be allowed to connect to the following IP addresses in order to join the Teleport cluster.
```

Author: Went with "cloud-hosted" for consistency with the other wording. It's important that self-hosted customers understand that this is not necessary (and less secure) for their setup.

```
3.7.23.103/32
3.67.31.207/32
3.109.188.166/32
13.215.3.254/32
18.136.170.204/32
18.197.230.105/32
18.228.73.158/32
35.82.240.238/32
44.198.252.22/32
44.213.172.215/32
44.217.250.22/32
50.112.183.104/32
52.59.15.95/32
52.66.6.236/32
52.67.142.215/32
52.220.247.39/32
54.94.211.77/32
54.185.13.106/32
```

> This list may be used to allowlist outbound network connections from Teleport Agents to Cloud-hosted Teleport Enterprise.

Reviewer: ```suggestion
This list may be used to allowlist outbound network connections from Teleport Agents to cloud-hosted Teleport Enterprise.
```

Allowlisting these IPs is not a required or recommended configuration for Teleport Agents, but it may be useful in environments that require restrictions to outbound network connections.
For example, this list may be used to configure firewalls that sit in front of Teleport Agents, when those firewalls block access to the public internet by default.

<Notice type="warning">
Note that IP addresses may be added or removed from the above list over time.
</Notice>

When this list is modified, we will provide at least two weeks notice by:

> 1. Updating the Changelog below.

Reviewer: What will this look like during the next update? For an update two weeks from today, would it looks like:

```
- 2024-06-24: List updated
- 2024-06-06: List published
```

If updating the docs is part of the IP allowlist update workflow, would it make sense to include a `Notice` containing the date of the next allowlist update at the top of the page instead?

Author: It would be more descriptive, and less frequent, e.g.,
```
- 2026-06-24: 127.1.2.3/32 added to list.
- 2024-06-06: List published
```

Example of a similar workflow: https://www.cloudflare.com/ips/

> would it make sense to include a Notice containing the date of the next allowlist update at the top of the page instead

For security and process reasons, nothing needs to be announced before the list is updated. We could add a notice when we update the list, to call attention to the recent change.

Reviewer: That makes sense, thanks!

> 1. Notifying Cloud-hosted Teleport Enterprise customers via email.

Reviewer: ```suggestion
1. Notifying cloud-hosted Teleport Enterprise customers via email.
```
